client/packages/common/src/intl/utils/DateUtils.ts

```diff
@@ -83,13 +83,14 @@ export const DateUtils = {
   addCurrentTime,
   getDateOrNull: (
     date?: Date | string | null,
-    format?: string
+    format?: string,
+    options?: Parameters<typeof parse>[3]
```

Comment:
Might be overkill, but I wonder if we should tighten the type definitions here to ensure that if a string date is provided, then `format` and `options` (with `locale`) must also be provided (but optional if date is `Date`).

Might make the input type ridiculously complicated though, so only if you think its worth it.

Reply:
the `options` is only required if you have a format which uses locale specific items (such as `P` or `p` in this case) so I wouldn't want to make that required.
I like the idea, so had a look - and can make the type of `format` depend on the `date` type, something like:

```
  getDateOrNull: <T extends Date | string | null | undefined>(
    date: T,
    format: T extends string ? string : string | undefined,
    options?: Parameters<typeof parse>[3]
  ): Date | null => {
```

but.. couldn't see how to have format as optional, only how to have the type change i.e. `string | undefined`. the format paramater still has to be specified, and that results in lots of code changes to add in `getDateOrNull(date, undefined)` which isn't great.